### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by DmitryKolomeets**

### CODE REVIEW

File: src/lesson4/Car.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    private int mileage;
+    private int fuel;
+    private int speed;
+    private int coolantLevel; // Litres
+    private String color;
+    private String engine;
+
+    public Car() {
+    } // Constructor without parameters
```
Comment (shaitanaleksander) [inline src/lesson4/Car.java]: 👌

File: src/lesson4/Car.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        this.color = color;
+        this.engine = engine;
+    }
+
+    public int getMileage() {
+        return mileage;
+    }
+
+    public void setMileage(int dailyMileage) {
+        if (dailyMileage > 0 && dailyMileage <= 2000)  // 2000 daily norm for mileage
```
Comment (shaitanaleksander) [inline src/lesson4/Car.java]: 2000 looks like constant, you can create class variable something like: int dailyMileageMax = 2000; and use it here

File: src/lesson4/Car.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+            System.out.println("Wrong mileage. The indicator remained unchanged. ");
+        }
+    }
+
+    public int getFuel() {
+        return fuel;
+    }
+
+    public void setFuel(int refueling) {
+        if (refueling > 0 && refueling <= (60 - fuel))    // 60 - car tank limit
```
Comment (shaitanaleksander) [inline src/lesson4/Car.java]: int carTankLimit = 60 vill be better as a lass constant

File: src/lesson4/Car.java (hunk 2 above)
Comment (DmitryKolomeets) [reply src/lesson4/Car.java]: Thanks for the recommendation. I'll make changes to the code.

File: src/lesson4/Car.java (hunk 3 above)
Comment (DmitryKolomeets) [reply src/lesson4/Car.java]: Ок.

File: src/lesson4/FreightCar.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Override
+    public boolean isReadyToService() {
+        if (distanceOnService > kilometersToMaintenance) {
+            return true;
+        } else {
+            return false;
+        }
+    }
```
Comment (shaitanaleksander) [inline src/lesson4/FreightCar.java:21]: not a mistake but you can use ternary operator her

**Review by shaitanaleksander — APPROVED** [review]

**Pull request merged**